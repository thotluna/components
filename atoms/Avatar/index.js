export { default } from './Avatar'
export { options } from './constants'
export { default as styles } from './Avatar.module.css'
