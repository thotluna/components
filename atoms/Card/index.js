export { default } from './Card'
export { options } from './constants'
export { default as styles } from './Card.module.css'
