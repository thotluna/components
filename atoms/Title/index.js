export { default } from './Title'
export { options } from './constants'
export { default as styles } from './Title.module.css'
