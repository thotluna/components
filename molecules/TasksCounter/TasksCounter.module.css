.container-counter {
  width: 100%;
}

.messages {
  display: flex;
  width: 100%;
  align-items: center;
  justify-content: space-between;
}
