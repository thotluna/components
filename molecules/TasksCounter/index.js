export { default } from './TasksCounter'
export { options } from './constants'
export { default as styles } from './TasksCounter.module.css'
