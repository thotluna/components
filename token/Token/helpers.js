import toKebabCase from '../../utils/toKebalCase'

export const getTokenItems = (tokens, tokenKey) => (group) => {
  if (group) {
    return Object.keys(tokens[group]).map((key) => ({
      label: `--${toKebabCase(tokenKey)}-${group}-${toKebabCase(key)}`,
      value: tokens[group][key],
    }))
  }

  return Object.keys(tokens).map((key) => ({
    label: `--${toKebabCase(tokenKey)}-${toKebabCase(key)}`,
    value: tokens[key],
  }))
}
