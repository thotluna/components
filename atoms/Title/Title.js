import React from 'react'
import PropTypes from 'prop-types'
import styles from './Title.module.css'
import { options } from './constants'
import { getClasses } from '../../helpers/styles'

export const Title = ({ children, color, size }) => {
  const getStyles = getClasses(styles)({ color })
  return (
    <>
      {size === 'lg' && (
        <h2 className={getStyles('title', ['color'])}>{children}</h2>
      )}
      {size === 'md' && (
        <h3 className={getStyles('title', ['color'])}>{children}</h3>
      )}
      {size === 'sm' && (
        <h4 className={getStyles('title', ['color'])}>{children}</h4>
      )}
    </>
  )
}

Title.propTypes = {
  children: PropTypes.node,
  color: PropTypes.oneOf(options.colors),
  size: PropTypes.oneOf(options.sizes),
}

Title.defaultProps = {
  color: 'base',
  size: 'sm',
}

export default Title
