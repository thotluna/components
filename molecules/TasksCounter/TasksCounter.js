import React from 'react'
import PropTypes from 'prop-types'
import styles from './TasksCounter.module.css'
import { options } from './constants'

import SeparatorLine from '../../atoms/SeparatorLine'
import Title from '../../atoms/Title'

import { getClasses } from '../../helpers/styles'

export const TasksCounter = ({ message, color, current, total }) => {
  const counter = `${current}/${total}`
  const getStyles = getClasses(styles)({})
  return (
    <div className={getStyles('container-counter')}>
      <div className={getStyles('messages')}>
        <Title color={color}>{message}</Title>
        <Title color={color}>{counter}</Title>
      </div>
      <SeparatorLine title="Task Counter" color={color} />
    </div>
  )
}

TasksCounter.propTypes = {
  message: PropTypes.string.isRequired,
  color: PropTypes.oneOf(options.colors),
  current: PropTypes.number.isRequired,
  total: PropTypes.number.isRequired,
}

TasksCounter.defaultProps = {
  message: 'test',
  current: 10,
  total: 10,
}

export default TasksCounter
