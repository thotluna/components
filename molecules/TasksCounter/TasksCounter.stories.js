import React from 'react'
import TasksCounter from '.'
import { options } from '../../atoms/Title'
import { getVariusTemplate, getOptionsArgTypes } from '../../helpers/storybook'

const Template = (args) => <TasksCounter {...args} />
const ListTemplate = getVariusTemplate(TasksCounter)

export default {
  title: 'Molecules/TasksCounter',
  component: TasksCounter,
  args: {
    message: 'Task resolved',
    current: 5,
    total: 10,
    color: 'primary',
  },
  argTypes: {
    message: { control: 'text' },
    color: getOptionsArgTypes(options.colors),
    current: { control: 'number' },
    total: { control: 'number' },
  },
}

export const Default = Template.bind({})

export const Colors = ListTemplate.bind({})
Colors.args = { items: options.colors.map((color) => ({ color })) }
