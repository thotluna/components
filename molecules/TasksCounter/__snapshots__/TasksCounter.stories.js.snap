// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Molecules/TasksCounter Colors 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
    }
  }
>
  <div
    className="container-counter"
  >
    <div
      className="messages"
    >
      <h3
        className="container color-base"
      >
        Task resolved
      </h3>
      <h3
        className="container color-base"
      >
        5/10
      </h3>
    </div>
    <div
      className="container-separator color-base"
    >
      <hr
        className="separator-line color-base size-sm"
        title="Task Counter"
      />
    </div>
  </div>
  <div
    className="container-counter"
  >
    <div
      className="messages"
    >
      <h3
        className="container color-primary"
      >
        Task resolved
      </h3>
      <h3
        className="container color-primary"
      >
        5/10
      </h3>
    </div>
    <div
      className="container-separator color-primary"
    >
      <hr
        className="separator-line color-primary size-sm"
        title="Task Counter"
      />
    </div>
  </div>
  <div
    className="container-counter"
  >
    <div
      className="messages"
    >
      <h3
        className="container color-secondary"
      >
        Task resolved
      </h3>
      <h3
        className="container color-secondary"
      >
        5/10
      </h3>
    </div>
    <div
      className="container-separator color-secondary"
    >
      <hr
        className="separator-line color-secondary size-sm"
        title="Task Counter"
      />
    </div>
  </div>
  <div
    className="container-counter"
  >
    <div
      className="messages"
    >
      <h3
        className="container color-tertiary"
      >
        Task resolved
      </h3>
      <h3
        className="container color-tertiary"
      >
        5/10
      </h3>
    </div>
    <div
      className="container-separator color-tertiary"
    >
      <hr
        className="separator-line color-tertiary size-sm"
        title="Task Counter"
      />
    </div>
  </div>
  <div
    className="container-counter"
  >
    <div
      className="messages"
    >
      <h3
        className="container color-inverted"
      >
        Task resolved
      </h3>
      <h3
        className="container color-inverted"
      >
        5/10
      </h3>
    </div>
    <div
      className="container-separator color-inverted"
    >
      <hr
        className="separator-line color-inverted size-sm"
        title="Task Counter"
      />
    </div>
  </div>
</div>
`;

exports[`Storyshots Molecules/TasksCounter Default 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
    }
  }
>
  <div
    className="container-counter"
  >
    <div
      className="messages"
    >
      <h3
        className="container color-primary"
      >
        Task resolved
      </h3>
      <h3
        className="container color-primary"
      >
        5/10
      </h3>
    </div>
    <div
      className="container-separator color-primary"
    >
      <hr
        className="separator-line color-primary size-sm"
        title="Task Counter"
      />
    </div>
  </div>
</div>
`;
